launch_ros/launch_ros/actions/node.py

```diff
@@ -408,6 +453,10 @@ def execute(self, context: LaunchContext) -> Optional[List[Action]]:
             ros_specific_arguments['name'] = '__node:={}'.format(self.__expanded_node_name)
         if self.__expanded_node_namespace != '':
             ros_specific_arguments['ns'] = '__ns:={}'.format(self.__expanded_node_namespace)
+        node_info = NodeActionExtension.NodeInfo(self.__package, self.__node_executable, self.node_name)
+        for extension in self.__extensions.values():
+            ros_specific_arguments = extension.execute(context, ros_specific_arguments, node_info)
```

Comment:
@gbiggs why not passing the node action itself instead of this ad-hoc data structure?

Reply:
It seemed cleaner to pass in a minimal data structure rather than have extensions need to know about the entire `Node` action class and its internal variables.

Reply:
I see. I will point out that this data in particular (e.g. the package name) could be exposed via public API. You wouldn't have to access internal implementations details from the extension in that case.

Reply:
I've added public API to expose the package and executable, and removed the `node_info` structure in 2d9f3d1.